proposal.md

```diff
@@ -0,0 +1,52 @@
+# Preamble
+
+Title: nft-barter Proposal document 
+
+## Simple Summary
+This is proposal document to describe nft-barter.
+
+## Abstract 
+This nft-barter is an open market where you can mint, buy, sell and swap your nft's within collection. The mint, buy and sell funtionality will be
+the standard mplementation ERC-721 with some add-ons. We would have our implementation of swap nft where user can list item for swap, propose swap and
+execute swap and pay-receive.
+
+## Motivation
+This idea is a widespread use case of trading / exchanging goods in both virtual and real world
+
+#### Real world example: 
+You have an iphone13 and you decide to switch to android,
+  1. You put up your iphone available for exchange with samsung on an online marketplace where you also find someone having a samsung s20 willing to swap with iphone.
```

Comment:
Here for my Iphone, I'll select the other samsung device specifically or would I create some open proposal and check for some other open proposals for selling samsung?

Reply:
There is an option for an open proposal as well.  It's specified in case 2 under implementation.